### ISSUE TITLE
Babel plugin

### ISSUE THREAD

**Issue by alippai** [angular/zone.js#1052]
As you can't hook async+await now (like you can the Promise/statemachine transpilations) is there a babel plugin which doesn't transpile the async+await syntax, but adds the required zone.js calls at transpile time?

**Comment by JiaLiPassion** [comment]
@alippai , could you describe your requirement more clearly? now typescript can transpile the `async/await` to `javascript Promise`. I am not sure about the `babel plugin`.

**Comment by alippai** [comment]
I want to leave it as async+await, but append a zonejs "async_call_completed()" call after each await.
```
async function a() {
  await b();
}
```
decorated into:
```
async function a() {
  await b();
  zonejs.async_call_completed();
}
```

**Comment by JiaLiPassion** [comment]
what do you mean `async_call_completed()`? could you check this one https://github.com/angular/zone.js/pull/795

**Comment by alippai** [comment]
`async_call_completed()` should be some callback which should be called after a change happened. I'm not familiar with zone.js task API/internals, sorry.
You monkey patch e.g. the setTimeout() now, so if the timer run, you invoke some zone.js internals resulting in angular change detection.
I'm looking for a way to transpile the:
```
async function a() {
  await b();
}
```
to trigger the Zone.js based Angular change detections, but without a complete ES8->ES5 transpilation.
Basically the same what you linked, but instead of a runtime async_hook a compile time one.

**Comment by JiaLiPassion** [comment]
@alippai , I see, without `async_hooks`, I don't think `zone.js` can do that, because even we can add such kind of `aysnc_call_completed()` with transpile, `zone.js` need to remember the `Zone Context` information when the async task was scheduled, but currently, without `async_hooks`, this is not possible.

**Comment by alippai** [comment]
I see, thank you!
